Unable to build LVGL with picolibc from source/module

### Describe the bug

When attempting to build LVGL with the Picolibc source module things fail to compile badly.


### Regression

- [x] This is a regression.

### Steps to reproduce

1. mkdir build && cd build
2. cmake -GNinja -DBOARD=native_sim ../samples/modules/lvgl/demos/ -DCONFIG_PICOLIBC=y
3. ninja
4. see errors

### Relevant log output

```shell
In file included from zephyr/build/modules/picolibc/picolibc/include/ssp/string.h:35,
                 from zephyr/build/modules/picolibc/picolibc/include/string.h:240,
                 from zephyr/modules/lvgl/include/lv_conf.h:12,
                 from modules/lib/gui/lvgl/demos/music/assets/../../../src/lv_conf_internal.h:56,
                 from modules/lib/gui/lvgl/demos/music/assets/../../../src/lv_init.h:16,
                 from modules/lib/gui/lvgl/demos/music/assets/../../../lvgl.h:21,
                 from modules/lib/gui/lvgl/demos/music/assets/../../lv_demos.h:16,
                 from modules/lib/gui/lvgl/demos/music/assets/../lv_demo_music.h:16,
                 from modules/lib/gui/lvgl/demos/music/assets/img_lv_demo_music_cover_3.c:1:
zephyr/modules/lvgl/include/lv_conf.h:41:35: error: expected declaration specifiers or ‘...’ before numeric constant
   41 | #define lv_memcpy                 memcpy
      |                                   ^~~~~~
modules/lib/gui/lvgl/demos/music/assets/../../../src/stdlib/lv_string.h:39:8: note: in expansion of macro ‘lv_memcpy’
   39 | void * lv_memcpy(void * dst, const void * src, size_t len);
      |        ^~~~~~~~~
zephyr/modules/lvgl/include/lv_conf.h:41:35: error: expected ‘)’ before ‘!=’ token
   41 | #define lv_memcpy                 memcpy
      |                                   ^~~~~~
modules/lib/gui/lvgl/demos/music/assets/../../../src/stdlib/lv_string.h:39:8: note: in expansion of macro ‘lv_memcpy’
   39 | void * lv_memcpy(void * dst, const void * src, size_t len);
      |        ^~~~~~~~~

(between many others)
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- Linux Ubuntu 24.04
- host gcc 14.2
- zephyr version main as of now: 52bf8fbf61d87b59a7ab417cf0edcbf9dd8689fa

### Additional Context

Related to
* #92898
* #93943

Note the picolibc source module and the one prebuilt with the SDK are not the same picolibc version in general

Note this was worsened by 47c4e3d813da06980d24f9c5718f3be82569b814, but it was failing before, since before v4.1.0, similarly on snprintf and vsnprintf:
```
In file included from build/modules/picolibc/picolibc/include/stdio.h:573,
                 from include/zephyr/sys/printk.h:65,
                 from include/zephyr/kernel_includes.h:35,
                 from include/zephyr/kernel.h:17,
                 from samples/modules/lvgl/demos/src/main.c:7:
modules/lvgl/include/lv_conf.h:30:35: error: expected declaration specifiers or ‘...’ before numeric constant
   30 | #define lv_snprintf               snprintf
      |                                   ^~~~~~~~
/scratch/zephyrproject/modules/lib/gui/lvgl/src/../src/stdlib/lv_sprintf.h:52:5: note: in expansion of macro ‘lv_snprintf’
   52 | int lv_snprintf(char * buffer, size_t count, const char * format, ...);
      |     ^~~~~~~~~~~
In file included from build/modules/picolibc/picolibc/include/ssp/string.h:35,
                 from build/modules/picolibc/picolibc/include/string.h:240,
                 from include/zephyr/sys/util.h:31,
                 from include/zephyr/sys/atomic.h:18,
                 from include/zephyr/kernel_includes.h:25:
modules/lvgl/include/lv_conf.h:30:35: error: expected declaration specifiers or ‘...’ before ‘__builtin_dynamic_object_size’
   30 | #define lv_snprintf               snprintf
      |                                   ^~~~~~~~
/scratch/zephyrproject/modules/lib/gui/lvgl/src/../src/stdlib/lv_sprintf.h:52:5: note: in expansion of macro ‘lv_snprintf’
   52 | int lv_snprintf(char * buffer, size_t count, const char * format, ...);
      |     ^~~~~~~~~~~
modules/lvgl/include/lv_conf.h:31:35: error: expected declaration specifiers or ‘...’ before numeric constant
   31 | #define lv_vsnprintf              vsnprintf
      |                                   ^~~~~~~~~
/scratch/zephyrproject/modules/lib/gui/lvgl/src/../src/stdlib/lv_sprintf.h:54:5: note: in expansion of macro ‘lv_vsnprintf’
   54 | int lv_vsnprintf(char * buffer, size_t count, const char * format, va_list va);
      |     ^~~~~~~~~~~~
modules/lvgl/include/lv_conf.h:31:35: error: expected declaration specifiers or ‘...’ before ‘__builtin_dynamic_object_size’
   31 | #define lv_vsnprintf              vsnprintf
      |                                   ^~~~~~~~~
/scratch/zephyrproject/modules/lib/gui/lvgl/src/../src/stdlib/lv_sprintf.h:54:5: note: in expansion of macro ‘lv_vsnprintf’
   54 | int lv_vsnprintf(char * buffer, size_t count, const char * format, va_list va);
      |     ^~~~~~~~~~~~
```
Neither of this issues were present on the 4.0 release

--- a/modules/lvgl/CMakeLists.txt
+++ b/modules/lvgl/CMakeLists.txt
@@ -16,6 +16,7 @@ zephyr_include_directories(${LVGL_DIR}/src/)
 zephyr_include_directories(include)
 
 zephyr_compile_definitions(LV_CONF_INCLUDE_SIMPLE=1)
+zephyr_library_compile_definitions(_POSIX_C_SOURCE=200809L)
 zephyr_compile_definitions(LV_CONF_PATH="${CMAKE_CURRENT_SOURCE_DIR}/include/lv_conf.h")
 
 zephyr_library_sources(
@@ -309,6 +310,8 @@ zephyr_library_sources(
     ${LVGL_DIR}/src/stdlib/builtin/lv_sprintf_builtin.c
     ${LVGL_DIR}/src/stdlib/builtin/lv_string_builtin.c
     ${LVGL_DIR}/src/stdlib/builtin/lv_tlsf.c
+    ${LVGL_DIR}/src/stdlib/clib/lv_string_clib.c
+    ${LVGL_DIR}/src/stdlib/clib/lv_sprintf_clib.c
 
     ${LVGL_DIR}/src/stdlib/clib/lv_mem_core_clib.c
     ${LVGL_DIR}/src/stdlib/clib/lv_sprintf_clib.c

--- a/modules/lvgl/include/lv_conf.h
+++ b/modules/lvgl/include/lv_conf.h
@@ -12,7 +12,9 @@
 
 /* Memory manager settings */
 
-#define LV_USE_STDLIB_MALLOC LV_STDLIB_CUSTOM
+#define LV_USE_STDLIB_MALLOC  LV_STDLIB_CUSTOM
+#define LV_USE_STDLIB_STRING  LV_STDLIB_CLIB
+#define LV_USE_STDLIB_SPRINTF LV_STDLIB_CLIB
 
 #if defined(CONFIG_LV_Z_MEM_POOL_HEAP_LIB_C)
 #define LV_STDLIB_INCLUDE "stdlib.h"
@@ -26,9 +28,6 @@
 #define lv_free_core      lvgl_free
 #endif
 
-/* Misc settings */
-#define lv_snprintf               snprintf
-#define lv_vsnprintf              vsnprintf
 #define LV_ASSERT_HANDLER         __ASSERT_NO_MSG(false);
 #define LV_ASSERT_HANDLER_INCLUDE "zephyr/sys/__assert.h"
 

